tests/scripts/thread-cert/pktverify/bytes.py

```diff
@@ -0,0 +1,149 @@
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+import sys
+from typing import Union
+
+
+class Bytes(bytearray):
+    """Bytes represents a byte array which is able to handle strings of flexible formats"""
+
+    def __init__(self, s: Union[str, bytearray, 'Bytes']):
+        if isinstance(s, str):
+            try:
+                s = Bytes._parse_compact(s)
+            except ValueError:
+                try:
+                    s = Bytes._parse_octets(s)
+                except ValueError:
+                    s = Bytes._parse_hextets(s)
+
+        super(Bytes, self).__init__(s)
+
+    def __hash__(self):
+        return hash(bytes(self))
+
+    def __repr__(self):
+        return '%s(%r)' % (self.__class__.__name__, self.format_compact())
+
+    def format_compact(self) -> str:
+        return ''.join('%02x' % b for b in self)
+
+    def format_octets(self) -> str:
+        return ':'.join('%02x' % b for b in self)
+
+    def format_hextets(self) -> str:
+        assert len(self) % 2 == 0, self.format_octets()
+        return ':'.join('%04x' % (self[i] * 256 + self[i + 1]) for i in range(0, len(self), 2))
+
+    __str__ = format_octets
+
+    @staticmethod
+    def _parse_compact(s: str) -> bytearray:
+        try:
+            assert len(s) % 2 == 0
+            return bytearray(int(s[i:i + 2], 16) for i in range(0, len(s), 2))
+        except Exception:
+            raise ValueError(s)
+
+    @staticmethod
+    def _parse_octets(s: str) -> bytearray:
+        try:
+            assert len(s) % 3 == 2 or not s
+            if not s:
+                return bytearray(b"")
+
+            return bytearray(int(x, 16) for x in s.split(':'))
+        except Exception:
+            raise ValueError(s)
+
+    @staticmethod
+    def _parse_hextets(s) -> bytearray:
+        try:
+            assert len(s) % 5 == 4 or not s
+            if not s:
+                return bytearray(b"")
+
+            return bytearray(int(x[i:i + 2], 16) for x in s.split(':') for i in (0, 2))
+        except Exception:
+            raise ValueError(s)
+
+    def __getitem__(self, item) -> Union['Bytes', int]:
+        """
+        Get self[item]
+        :param item: index or slice to retrieve
+        :returns: the byte value at specified index or sub `Bytes` if item is slice
+        """
+        x = super(Bytes, self).__getitem__(item)
+        if isinstance(x, bytearray):
+            return Bytes(x)
+        else:
+            return x
+
+    def __eq__(self, other: Union[str, 'Bytes']):
+        """
+        Check if bytes is equal to other
+        """
+        if other is None:
+            return False
+        elif not isinstance(other, Bytes):
+            other = self.__class__(other)
+
+        eq = super(Bytes, self).__eq__(other)
+        print("[%r %s %r]" % (self, "==" if eq else "!=", other), file=sys.stderr)
```

Comment:
It's printed on purpose. I tends to keep this log because they are useful when you are writing the verification script. It tells you how the verification process goes and what conditions it has calculated.